pkg/login/social/social.go

```diff
@@ -372,11 +372,32 @@ func (s *SocialBase) SupportBundleContent(bf *bytes.Buffer) error {
 	return nil
 }
 
-func (s *SocialBase) extractRoleAndAdmin(rawJSON []byte, groups []string, legacy bool) (org.RoleType, bool) {
+func (s *SocialBase) extractRoleAndAdmin(rawJSON []byte, groups []string, legacy bool) (org.RoleType, bool, error) {
 	if s.roleAttributePath == "" {
-		return s.defaultRole(legacy), false
+		if s.roleAttributeStrict {
+			return "", false, errRoleAttributePathNotSet.Errorf("role_attribute_path not set and role_attribute_strict is set")
+		}
+		return s.defaultRole(), false, nil
+	}
+
+	role, gAdmin := s.searchRole(rawJSON, groups)
+	if !role.IsValid() {
+		if s.roleAttributeStrict {
+			return "", false, errRoleAttributeStrictViolation.Errorf("idP did not return a role attribute, but role_attribute_strict is set")
+		}
+
+		if role != "" {
+			return "", false, errInvalidRole.Errorf("invalid role: %s", role)
```

Comment:
Note: looks like we already check that the role is valid when we read the settings: https://github.com/grafana/grafana/blob/main/pkg/setting/setting.go#L1641 But I guess it doesn't hurt to do it twice.